pkg/xds/generator/egress/generator.go

```diff
@@ -0,0 +1,189 @@
+package egress
+
+import (
+	"github.com/pkg/errors"
+
+	mesh_proto "github.com/kumahq/kuma/api/mesh/v1alpha1"
+	"github.com/kumahq/kuma/pkg/core"
+	core_mesh "github.com/kumahq/kuma/pkg/core/resources/apis/mesh"
+	core_xds "github.com/kumahq/kuma/pkg/core/xds"
+	xds_context "github.com/kumahq/kuma/pkg/xds/context"
+	envoy_common "github.com/kumahq/kuma/pkg/xds/envoy"
+	envoy_listeners "github.com/kumahq/kuma/pkg/xds/envoy/listeners"
+	envoy_names "github.com/kumahq/kuma/pkg/xds/envoy/names"
+)
+
+const (
+	EgressProxy = "egress-proxy"
+
+	// OriginEgress is a marker to indicate by which ProxyGenerator resources
+	// were generated.
+	OriginEgress = "egress"
+)
+
+var (
+	log = core.Log.WithName("xds").WithName("generator").WithName("egress")
+)
+
+type Destination struct {
+	Destination envoy_common.Tags
+	Weight      uint32
+
+	IsExternalService bool
+}
+
+type Listener struct {
+	Address      string
+	Port         uint32
+	ResourceName string
+}
+
+// Resources tracks partially-built xDS resources that can be updated
+// by multiple zoneegress generators.
+type Resources struct {
+	Listener *envoy_listeners.ListenerBuilder
```

Comment:
Do you forsee any other resources added here? If not, can we just put this into ResoruceInfo?

Reply:
removed